File: 백준/Bronze/13304. 방 배정/방 배정.py
N, K = map(int, input().split())
student = [[0] * 3 for _ in range(2)]

for _ in range(N):
    S, Y = map(int, input().split())
    student[S][(Y - 1) // 2] += 1

print(sum((cnt + K - 1) // K for row in student for cnt in row))

File: 백준/Bronze/13304. 방 배정/방 배정.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : 'run/input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

const [N, K] = input[0].split(' ').map(Number);
const student = Array.from({ length: 2 }, () => [0, 0, 0]);

for (let i = 1; i <= N; i++) {
  const [S, Y] = input[i].split(' ').map(Number);
  student[S][Math.floor((Y - 1) / 2)]++;
}

const room = student.flat().reduce((sum, cnt) => sum + Math.ceil(cnt / K), 0);

console.log(room);
